src/c_api/factorization/cholesky.h

```diff
@@ -0,0 +1,118 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2023, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#pragma once
+
+#include <mpi.h>
+
+#include <pika/init.hpp>
+
+#include <dlaf/factorization/cholesky.h>
+#include <dlaf/matrix/matrix.h>
+#include <dlaf/matrix/matrix_mirror.h>
+#include <dlaf_c/desc.h>
+#include <dlaf_c/grid.h>
+
+#include "../blacs.h"
+#include "../grid.h"
+#include "../utils.h"
+
+template <typename T>
+int cholesky(int dlaf_context, char uplo, T* a, DLAF_descriptor dlaf_desca) noexcept {
+  try {
+    using MatrixMirror = dlaf::matrix::MatrixMirror<T, dlaf::Device::Default, dlaf::Device::CPU>;
+
+    DLAF_ASSERT(dlaf_desca.i == 1, dlaf_desca.i);
+    DLAF_ASSERT(dlaf_desca.j == 1, dlaf_desca.j);
+
+    pika::resume();
+
+    auto dlaf_uplo = dlaf_uplo_from_char(uplo);
+
+    auto communicator_grid = dlaf_grids.at(dlaf_context);
+
+    dlaf::GlobalElementSize matrix_size(dlaf_desca.m, dlaf_desca.n);
+    dlaf::TileElementSize block_size(dlaf_desca.mb, dlaf_desca.nb);
+
+    dlaf::comm::Index2D src_rank_index(dlaf_desca.isrc, dlaf_desca.jsrc);
+
+    dlaf::matrix::Distribution distribution(matrix_size, block_size, communicator_grid.size(),
+                                            communicator_grid.rank(), src_rank_index);
+
+    dlaf::matrix::LayoutInfo layout = colMajorLayout(distribution, dlaf_desca.ld);
+
+    dlaf::matrix::Matrix<T, dlaf::Device::CPU> matrix_host(std::move(distribution), layout, a);
+
+    {
+      MatrixMirror matrix(matrix_host);
+
+      dlaf::factorization::cholesky<dlaf::Backend::Default, dlaf::Device::Default, T>(communicator_grid,
+                                                                                      dlaf_uplo,
+                                                                                      matrix.get());
+    }  // Destroy mirror
+
+    matrix_host.waitLocalTiles();
+
+    pika::suspend();
+
+    return 0;
+  }
+  catch (std::exception& e) {
+    std::cerr << e.what() << '\n';
+    return -1;
+  }
```

Comment:
I think most exceptions will inherit from `std::exception` but they're not guaranteed to do so. We could either:
- do what you do now, and non-`std::exception`s will terminate at the end of the function because it's `noexcept`, or
- add another catch-all catch:
```suggestion
  catch (const std::exception& e) {
    std::cerr << e.what() << '\n';
    return -1;
  }
  catch (...)
  {
    std::cerr << "blabla unknown exception thrown in dlaf_xxx\n";
    return -1;
  }
```
I think both are fine, but I'm slightly in favour of catching all exceptions and let the caller deal with the `-1`.